src/build-remote/build-remote.cc

```diff
@@ -286,23 +288,58 @@ static int main_build_remote(int argc, char * * argv)
         uploadLock = -1;
 
         auto drv = store->readDerivation(*drvPath);
-        auto outputHashes = staticOutputHashes(*store, drv);
-
-        // Hijack the inputs paths of the derivation to include all the paths
-        // that come from the `inputDrvs` set.
-        // We don’t do that for the derivations whose `inputDrvs` is empty
-        // because
-        // 1. It’s not needed
-        // 2. Changing the `inputSrcs` set changes the associated output ids,
-        //  which break CA derivations
-        if (!drv.inputDrvs.empty())
-            drv.inputSrcs = store->parseStorePathSet(inputs);
 
-        auto result = sshStore->buildDerivation(*drvPath, drv);
+        std::optional<BuildResult> optResult;
+
+        // Let's break this down
+        //
+        // ### Trust part
+        //
+        // ```
+        // std::optional trust = sshStore->isTrustedClient(); (!trust || *trust)
+        // ```
+        //
+        // If we don't know whether we are trusted (e.g. `ssh://`
+        // stores), we assume we are. This is necessary for backwards
+        // compat.
+        //
+        // ### Content-addressing part
+        //
+        // ```
+        // ...trustCond... || drv.type().isCA()
+        // ```
+        //
+        // See the very large comment in `case wopBuildDerivation:` in
+        // `src/libstore/daemon.cc` that explains the trust model here.
+        //
+        // This condition mirrors that: that code enforces the "rules";
+        // we do the best we can given those "rules".
+        if (std::optional trust = sshStore->isTrustedClient(); (!trust || *trust) || drv.type().isCA()) {
```

Comment:
Put the explanatory comments around the helper variables.
```suggestion
        std::optional trusted = sshStore->isTrustedClient();
        // for backward compatibility (use existing comments here)
        bool trustedOrLegacy = !trusted || *trusted;
        if (trustedOrLegacy || drv.type().isCA())  {
```